addons/gen.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+WD=$(dirname "$0")
+WD=$(cd "$WD"; pwd)
+
+set -eux
+
+# This script sets up the plain text rendered deployments for addons
+# See samples/addons/README.md for more information
+
+TEMPLATES="${WD}/../templates"
+DASHBOARDS="${WD}/dashboards"
+TMP=$(mktemp -d)
+
+# Set up prometheus
+helm template prometheus prometheus \
+  --namespace "replace-me-namespace" \
+  --version 13.2.1 \
+  --repo https://prometheus-community.github.io/helm-charts \
+  -f "${WD}/values-prometheus.yaml" \
+  > "${TEMPLATES}/prometheus.yaml"
+
+sed -i'.orig' 's/replace-me-namespace/{{ .Release.Namespace }}/g' "${TEMPLATES}/prometheus.yaml"
+sed -i'.orig' '1i\
+{{- if .Values.prometheus.enabled }}
+' "${TEMPLATES}/prometheus.yaml"
+sed -i'.orig' -e '$a\
+{{- end }}' "${TEMPLATES}/prometheus.yaml"
+rm "${TEMPLATES}/prometheus.yaml.orig"
```

Comment:
I would love if there was a simpler way to 
• Replace `replace-me-namespace` with `{{ .Release.Namespace }}`
• Top and tail the file with `{{- if .Values.prometheus.enabled }}` and `{{- end }}`

Reply:
Yeah can you add comments about what each search/replace is doing